langchain/vectorstores/rocksetdb.py

```diff
@@ -0,0 +1,308 @@
+
+from langchain.docstore.document import Document
+from langchain.embeddings.base import Embeddings
+from langchain.vectorstores.base import VectorStore
+
+logger = logging.getLogger(__name__)
+
+
+"""
+Everything below assumes `commons` Rockset workspace. TODO: Add support for workspace args.
+"""
+
+
+class RocksetDB(VectorStore):
+    """Wrapper arpund Rockset vector database.
+
+    To use, you should have the `rockset` python package installed. Note that to use this,
+    the collection being used must already exist in your Rockset instance.
+    You must also ensure you use a Rockset ingest transformation to apply `VECTOR_ENFORCE`
+    on the column being used to store `embedding_key` in the collection.
+    See: https://rockset.com/blog/introducing-vector-search-on-rockset/ for more details
+
+    Example:
+        .. code-block:: python
+
+            from langchain.vectorstores import Rockset
+            from langchain.embeddings.openai import OpenAIEmbeddings
+            import rockset
+
+            # Make sure you use the right host (region) for your Rockset instance
+            # and APIKEY has both read-write access to your collection.
+
+            rs = rockset.RocksetClient(host=rockset.Regions.use1a1, api_key="***")
+            collection_name = "langchain_demo"
+            embeddings = OpenAIEmbeddings()
+            vectorstore = Rockset(rs, collection_name, embeddings,
+                "description", "description_embedding")
+
+    """
+
+    def __init__(
+        self,
+        client: Any,
+        embeddings: Embeddings,
+        collection_name: str,
+        text_key: str,
+        embedding_key: str,
+    ):
+        """Initialize with Rockset client.
+        Args:
+            client: Rockset client object
+            collection: Rockset collection to insert docs / query
+            embeddings: Langchain Embeddings object to use to generate
+                        embedding for given text.
+            text_key: column in Rockset collection to use to store the text
+            embedding_key: column in Rockset collection to use to store the embedding.
+                           Note: We must apply `VECTOR_ENFORCE()` on this column via
+                           Rockset ingest transformation.
+
+        """
+        if not isinstance(client, RocksetClient):
+            raise ValueError(
+                f"client should be an instance of rockset.RocksetClient, "
+                f"got {type(client)}"
+            )
+        # TODO: check that `collection_name` exists in rockset. Create if not.
+        self._client = client
+        self._collection_name = collection_name
+        self._embeddings = embeddings
+        self._text_key = text_key
+        self._embedding_key = embedding_key
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        ids: Optional[List[str]] = None,
+        batch_size: int = 32,
```

Comment:
I wonder if mypy. Have you run their linters yet? If it doesn't complain, then I guess you don't need to change anything here.

Reply:
Ran their linter, there were issues around subclass having different signature, were you talking about that one? :D Regardless, all lint errors fixed now.